src/transaction.js

```diff
@@ -196,61 +300,38 @@ Transaction.fromBuffer = function (buffer, network = networks.bitcoin, __noStric
     tx.expiryHeight = readUInt32()
   }
 
+  if (coins.isZcash(network) && tx.version >= Transaction.ZCASH_SAPLING_VERSION) {
+    tx.valueBalance = readUInt64()
```

Comment:
Should this be `readInt64()`?

Reply:
yes, should be signed

![image](https://user-images.githubusercontent.com/1884338/45653523-2218b600-ba8d-11e8-9b3d-364255757228.png)